pkg/controller/daemon/daemoncontroller.go

```diff
@@ -899,11 +899,35 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		return false, false, false, err
 	}
 
+	// Do not fire InsufficientResourceError when there are intentional reasons
+	fireInsufficientResourceError := true
+	for _, r := range reasons {
+		switch reason := r.(type) {
+		case *predicates.PredicateFailureError:
+			// we try to partition predicates into two partitions here: intentional on the part of the operator and not.
+			switch reason {
+			// intentional
+			case
+				predicates.ErrNodeSelectorNotMatch,
+				predicates.ErrPodNotMatchHostName,
+				predicates.ErrNodeLabelPresenceViolated,
+				// this one is probably intentional since it's a workaround for not having
+				// pod hard anti affinity.
+				predicates.ErrPodNotFitsHostPorts,
+				// DaemonSet is expected to respect taints and tolerations
+				predicates.ErrTaintsTolerationsNotMatch:
+				fireInsufficientResourceError = false
```

Comment:
`InsufficientResourceError` is only an example to me; if nodeSelector did not match, daemonSet should not report other predicates events.

Anyway, I'd like to get other's comments on this :).